src/Sylius/Bundle/ShippingBundle/Form/Type/AmountType.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ShippingBundle\Form\Type;
+
+use Sylius\Bundle\MoneyBundle\Form\Type\MoneyType;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\GreaterThan;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Type;
+
+final class AmountType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('amount', MoneyType::class, [
+                'label' => false,
+                'currency' => $options['channel']->getBaseCurrency()->getCode(),
+                'constraints' => [
+                    new NotBlank(['groups' => ['sylius']]),
+                    new Type(['type' => 'integer', 'groups' => ['sylius']]),
+                    new GreaterThan(['value' => 0, 'groups' => ['sylius']]),
+                ],
+            ])
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setRequired('channel')
+            ->setAllowedTypes('channel', [ChannelInterface::class])
+
+            ->setDefaults([
+                'label' => static function (Options $options): string {
+                    return $options['channel']->getName();
```

Comment:
But you need the channel name to distinguish the amounts for different channels

Reply:
Yes, so we could inject 2 variables: `label` (would pass the channel's name) and `currency` (would pass the channel's base currency code). We even **cannot** use `ChannelInterfae` there, as it couples `ShippingBundle` with `Core` component 🏎️

Reply:
I moved this type and associated classes to the core bundle since we need the channel in the form type:

![image](https://user-images.githubusercontent.com/2412177/89758470-2c605500-dae8-11ea-8421-5ee00ac4c72d.png)